File: backend_app/app/models/prompt_constraints.py
from __future__ import annotations
from typing import Literal, Optional
from pydantic import BaseModel


class PromptConstraints(BaseModel):
    format: Optional[Literal["bullets", "prose", "table"]] = None
    max_items: Optional[int] = None
    max_words: Optional[int] = None
    max_words_per_item: Optional[int] = None
    required_elements: Optional[list[str]] = None
    tone: Optional[str] = None

File: frontend_app/src/features/prompt-management/types/promptConstraints.ts
export type ConstraintFormat = "bullets" | "prose" | "table";

export interface PromptConstraints {
  format?: ConstraintFormat;
  max_items?: number;
  max_words?: number;
  max_words_per_item?: number;
  required_elements?: Array<string>;
  tone?: string;
}

export type PromptConstraintsMap = Record<string, PromptConstraints>;
